<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>bugphyzz • bugphyzz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="bugphyzz">
<meta property="og:description" content="bugphyzz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bugphyzz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/bugphyzz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/attributes.html">Sources and Attributes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/bugphyzz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>bugphyzz</h1>
            <h3 data-toc-skip class="subtitle">A harmonized data
resource and software for enrichment analysis of microbial
physiologies</h3>
                        <h4 data-toc-skip class="author">Samuel
Gamboa</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:Samuel.Gamboa.Tuz@gmail.com" class="email">Samuel.Gamboa.Tuz@gmail.com</a>
      
                              <h4 data-toc-skip class="author">Levi
Waldron</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:levi.waldron@sph.cuny.edu" class="email">levi.waldron@sph.cuny.edu</a>
      
                  
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/bugphyzz/blob/HEAD/vignettes/bugphyzz.Rmd" class="external-link"><code>vignettes/bugphyzz.Rmd</code></a></small>
      <div class="hidden name"><code>bugphyzz.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><a href="https://github.com/waldronlab/bugphyzzExports" class="external-link">bugphyzz</a>
is an electronic database of standardized microbial annotations that
allows: 1. annotation of microbial taxa from multiple sources, and 2.
creation of microbial signatures based on shared attributes for bug set
enrichment analysis.</p>
</div>
<div class="section level2">
<h2 id="data-schema">Data schema<a class="anchor" aria-label="anchor" href="#data-schema"></a>
</h2>
<p>Annotations in bugphyzz represent the link between a taxon
(Bacteria/Archaea) and an attribute, as outlined in the data schema
provided below.</p>
<center>
<div class="float">
<img src="bugphyzz_data_schema.png" style="width:100.0%" alt="Data schema"><div class="figcaption"><strong>Data schema</strong></div>
</div>
</center>
<p><strong>Taxon-related</strong></p>
<p>Taxonomic data in bugphyzz is standardized according to the NCBI
taxonomy:</p>
<ol style="list-style-type: decimal">
<li>
<em>NCBI ID</em>: An integer representing the NCBI taxonomy ID
(taxid) associated with a taxon.</li>
<li>
<em>Rank</em>: A character string indicating the taxonomy rank,
including superkingdom, kingdom, phylum, class, order, family, genus,
species, or strain.</li>
<li>
<em>Taxon name</em>: A character string denoting the scientific name
of the taxon.</li>
</ol>
<p><strong>Attribute-related</strong></p>
<p>Attribute data is harmonized using ontology terms. Details of
attributes, ontology terms, and ontology libraries can be found in the
<a href="articles/attributes.html">Attribute and sources</a>
article.</p>
<ol start="4" style="list-style-type: decimal">
<li>
<em>Attribute</em>: A character string describing the name of a
trait that can be observed or measured.</li>
<li>
<em>Attribute type</em>: A character string indicating the data
type:
<ul>
<li>numeric: Attributes with numeric values (e.g., growth temperature:
25°C).</li>
<li>binary: Attributes with boolean values (e.g., butyrate-producing:
TRUE).</li>
<li>multistate-intersection: A set of related binary attributes (e.g.,
habitat).</li>
<li>multistate-union: Attributes with three or more values represented
as character strings (e.g., aerophilicity: aerobic, anaerobic, or
facultatively anaerobic).</li>
</ul>
</li>
<li>
<em>Attribute value</em>: The possible values that an attribute
could take, represented as character strings, booleans, or numbers.</li>
</ol>
<p><strong>Attribute value-related</strong></p>
<p>Metadata associated with attribute values:</p>
<ol start="7" style="list-style-type: decimal">
<li>
<em>Attribute source</em>: The source of the information.</li>
<li>
<em>Evidence</em>: The type of evidence supporting an annotation,
including:
<ul>
<li>EXP: Experiment</li>
<li>IGC: Inferred from genomic context</li>
<li>TAS: Traceable author statement</li>
<li>NAS: Non-traceable author statement</li>
<li>IBD: Inferred from biological aspect of descendant</li>
<li>ASR: Ancestral state reconstruction</li>
</ul>
</li>
<li>
<em>Support values</em>:
<ul>
<li>Frequency and Score: Confidence that a given taxon exhibits a trait
based on the curator’s knowledge or results of ASR or IBD.</li>
<li>Validation: Score from the <a href="https://github.com/waldronlab/taxPProValidation" class="external-link">10-fold
cross-validation analysis</a>. Matthews correlation coefficient (MCC)
for discrete attributes and R-squared for numeric attributes. Default
threshold value is 0.5 and above.</li>
<li>NSTI: Nearest sequence taxon index from <a href="https://doi.org/10.1038/nbt.2676" class="external-link">PICRUSt</a> or the <a href="https://cran.r-project.org/web/packages/castor/index.html" class="external-link">castor
package</a>. Relevant for numeric values only.</li>
</ul>
</li>
</ol>
<p><strong>Attribute source-related</strong></p>
<ol start="10" style="list-style-type: decimal">
<li>
<em>Confidence in curation</em>: A character string indicating the
confidence value of a source based on three criteria: 1) presence of a
source, 2) valid references, and 3) peer-reviewed curation. Valid
options include high, medium, or low, corresponding to satisfaction of
three, two, or one of these criteria.</li>
</ol>
<p><strong>Additional information</strong></p>
<ul>
<li><p>Description of <strong>sources</strong> and
<strong>attributes</strong>: <a href="https://waldronlab.io/bugphyzz/articles/attributes.html" class="uri">https://waldronlab.io/bugphyzz/articles/attributes.html</a></p></li>
<li><p>Description of ontology <strong>evidence</strong> codes: <a href="https://geneontology.org/docs/guide-go-evidence-codes/" class="external-link uri">https://geneontology.org/docs/guide-go-evidence-codes/</a></p></li>
<li><p>Description of <strong>frequency</strong> keywords and scores
were based on: <a href="https://grammarist.com/grammar/adverbs-of-frequency/" class="external-link uri">https://grammarist.com/grammar/adverbs-of-frequency/</a></p></li>
<li><p>IBD and ASR were performed with taxPPro: <a href="https://github.com/waldronlab/taxPPro" class="external-link uri">https://github.com/waldronlab/taxPPro</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="analysis-and-stats">Analysis and Stats<a class="anchor" aria-label="anchor" href="#analysis-and-stats"></a>
</h2>
<p>This vignette serves as an introduction to the basic functionalities
of bugphyzz. For a more in-depth analysis and detailed statistics
utilizing bugphyzz annotations, please visit: <a href="https://github.com/waldronlab/bugphyzzAnalyses" class="external-link uri">https://github.com/waldronlab/bugphyzzAnalyses</a></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"bugphyzz"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="import-bugphyzz-data">Import bugphyzz data<a class="anchor" aria-label="anchor" href="#import-bugphyzz-data"></a>
</h2>
<p>Load the bugphyzz package and additional packages for data
manipulation:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/bugphyzz" class="external-link">bugphyzz</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span></code></pre></div>
<p>bugphyzz data is imported using the <code>importBugphyzz</code>
function, resulting in a list of tidy data frames. Each data frame
corresponds to an attribute or a group of related attributes. This is
particularly evident in the case of the multistate-union type described
in the data schema above, where related attributes are grouped together
in a single data frame. Available attribute names can be inspected with
the <code>names</code> function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/importBugphyzz.html">importBugphyzz</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bp</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "aerophilicity"                        </span></span>
<span><span class="co">#&gt;  [2] "antimicrobial resistance"             </span></span>
<span><span class="co">#&gt;  [3] "arrangement"                          </span></span>
<span><span class="co">#&gt;  [4] "biosafety level"                      </span></span>
<span><span class="co">#&gt;  [5] "cogem pathogenicity rating"           </span></span>
<span><span class="co">#&gt;  [6] "disease association"                  </span></span>
<span><span class="co">#&gt;  [7] "gram stain"                           </span></span>
<span><span class="co">#&gt;  [8] "habitat"                              </span></span>
<span><span class="co">#&gt;  [9] "hemolysis"                            </span></span>
<span><span class="co">#&gt; [10] "shape"                                </span></span>
<span><span class="co">#&gt; [11] "spore shape"                          </span></span>
<span><span class="co">#&gt; [12] "animal pathogen"                      </span></span>
<span><span class="co">#&gt; [13] "antimicrobial sensitivity"            </span></span>
<span><span class="co">#&gt; [14] "biofilm formation"                    </span></span>
<span><span class="co">#&gt; [15] "butyrate-producing bacteria"          </span></span>
<span><span class="co">#&gt; [16] "extreme environment"                  </span></span>
<span><span class="co">#&gt; [17] "health associated"                    </span></span>
<span><span class="co">#&gt; [18] "host-associated"                      </span></span>
<span><span class="co">#&gt; [19] "hydrogen gas producing"               </span></span>
<span><span class="co">#&gt; [20] "lactate producing"                    </span></span>
<span><span class="co">#&gt; [21] "motility"                             </span></span>
<span><span class="co">#&gt; [22] "plant pathogenicity"                  </span></span>
<span><span class="co">#&gt; [23] "sphingolipid producing"               </span></span>
<span><span class="co">#&gt; [24] "spore formation"                      </span></span>
<span><span class="co">#&gt; [25] "coding genes"                         </span></span>
<span><span class="co">#&gt; [26] "genome size"                          </span></span>
<span><span class="co">#&gt; [27] "growth temperature"                   </span></span>
<span><span class="co">#&gt; [28] "length"                               </span></span>
<span><span class="co">#&gt; [29] "mutation rate per site per generation"</span></span>
<span><span class="co">#&gt; [30] "mutation rate per site per year"      </span></span>
<span><span class="co">#&gt; [31] "optimal ph"                           </span></span>
<span><span class="co">#&gt; [32] "width"</span></span></code></pre></div>
<p>Let’s take a glimpse at one of the data frames:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">bp</span><span class="op">$</span><span class="va">aerophilicity</span>, width <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 27,284</span></span>
<span><span class="co">#&gt; Columns: 12</span></span>
<span><span class="co">#&gt; $ NCBI_ID                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1042312, 117743, …</span></span>
<span><span class="co">#&gt; $ Taxon_name             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Armatimonadia", …</span></span>
<span><span class="co">#&gt; $ Rank                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "class", "class",…</span></span>
<span><span class="co">#&gt; $ Attribute              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "aerophilicity", …</span></span>
<span><span class="co">#&gt; $ Attribute_value        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "aerobic", "anaer…</span></span>
<span><span class="co">#&gt; $ Evidence               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "asr", "asr", "as…</span></span>
<span><span class="co">#&gt; $ Frequency              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "sometimes", "som…</span></span>
<span><span class="co">#&gt; $ Score                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.8781184, 0.6912…</span></span>
<span><span class="co">#&gt; $ Attribute_source       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ Confidence_in_curation <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ Attribute_type         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "multistate-inter…</span></span>
<span><span class="co">#&gt; $ Validation             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.73, 0.88, 0.73,…</span></span></code></pre></div>
<p>Compare the column names with the data schema described above.</p>
</div>
<div class="section level2">
<h2 id="create-microbial-signatures">Create microbial signatures<a class="anchor" aria-label="anchor" href="#create-microbial-signatures"></a>
</h2>
<p>bugphyzz’s primary function is to facilitate the creation of
microbial signatures, which are essentially lists of microbes sharing
specific taxonomy ranks and attribute values. Once the data frames
containing attribute information are imported, the
<code>makeSignatures</code> function can be employed to generate these
signatures. <code>makeSignatures</code> offers various filtering
options, including evidence, frequency, and minimum and maximum values
for numeric attributes. For more precise filtering requirements, users
can leverage standard data manipulation functions on the relevant data
frame, such as <code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter</a></code>.</p>
<p>Examples:</p>
<ul>
<li>Create signatures of taxon names at the genus level for the
aerophilicity attribute (discrete):</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aer_sigs_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="va">bp</span><span class="op">[[</span><span class="st">"aerophilicity"</span><span class="op">]</span><span class="op">]</span>, tax_id_type <span class="op">=</span> <span class="st">"Taxon_name"</span>, tax_level <span class="op">=</span> <span class="st">"genus"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">aer_sigs_g</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|aerobic`</span></span>
<span><span class="co">#&gt; [1] "Flavihumibacter"  "Parasegetibacter" "Ferruginibacter"  "Fimbriimonas"    </span></span>
<span><span class="co">#&gt; [5] "Camelimonas"      "Thiothrix"       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|anaerobic`</span></span>
<span><span class="co">#&gt; [1] "Microaerobacter"      "Desulfitispora"       "Desulfurispira"      </span></span>
<span><span class="co">#&gt; [4] "Pseudoflavonifractor" "Chromatium"           "Ectothiorhodospira"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|facultatively anaerobic`</span></span>
<span><span class="co">#&gt; [1] "Capnocytophaga" "Kistimonas"     "Trueperella"    "Telmatobacter" </span></span>
<span><span class="co">#&gt; [5] "Alishewanella"  "Allomuricauda"</span></span></code></pre></div>
<ul>
<li>Create signatures of taxon names at the species level for the growth
temperature attribute (numeric):</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gt_sigs_sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="va">bp</span><span class="op">[[</span><span class="st">"growth temperature"</span><span class="op">]</span><span class="op">]</span>, tax_id_type <span class="op">=</span> <span class="st">"Taxon_name"</span>,</span>
<span>  tax_level <span class="op">=</span> <span class="st">'species'</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">gt_sigs_sp</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`bugphyzz:growth temperature|hyperthermophile| &gt; 60 &amp; &lt;= 121`</span></span>
<span><span class="co">#&gt; [1] "Metallosphaera cuprina"      "Pyrococcus yayanosii"       </span></span>
<span><span class="co">#&gt; [3] "Methanothermobacter crinale" "Acidilobus aceticus"        </span></span>
<span><span class="co">#&gt; [5] "Thermanaerovibrio velox"     "Thermoanaerobacter italicus"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:growth temperature|mesophile| &gt; 25 &amp; &lt;= 45`</span></span>
<span><span class="co">#&gt; [1] "Ancylobacter aquaticus"          "Leptospira alexanderi"          </span></span>
<span><span class="co">#&gt; [3] "Halosaccharopolyspora lacisalsi" "Garritya polymorpha"            </span></span>
<span><span class="co">#&gt; [5] "Sporobacterium olearium"         "Borreliella lusitaniae"         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:growth temperature|psychrophile| &gt; -10.01 &amp; &lt;= 25`</span></span>
<span><span class="co">#&gt; [1] "Cryobacterium roopkundense"  "Hugenholtzia roseola"       </span></span>
<span><span class="co">#&gt; [3] "Halopseudomonas formosensis" "Occallatibacter riparius"   </span></span>
<span><span class="co">#&gt; [5] "Occallatibacter savannae"    "Pectinatus sottacetonis"    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:growth temperature|thermophile| &gt; 45 &amp; &lt;= 60`</span></span>
<span><span class="co">#&gt; [1] "Alicyclobacillus cellulosilyticus" "Defluviitoga tunisiensis"         </span></span>
<span><span class="co">#&gt; [3] "Fervidobacterium thailandense"     "Polycladomyces subterraneus"      </span></span>
<span><span class="co">#&gt; [5] "Desulfofalx alkaliphila"           "Marinitoga camini"</span></span></code></pre></div>
<ul>
<li>Create signatures with a custom threshold for the growth temperature
attribute (numeric):</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gt_sigs_mix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="va">bp</span><span class="op">[[</span><span class="st">"growth temperature"</span><span class="op">]</span><span class="op">]</span>, tax_id_type <span class="op">=</span> <span class="st">"Taxon_name"</span>,</span>
<span>  tax_level <span class="op">=</span> <span class="st">"mixed"</span>, min <span class="op">=</span> <span class="fl">0</span>, max <span class="op">=</span> <span class="fl">25</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">gt_sigs_mix</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`bugphyzz:growth temperature| &gt;=0 &amp; &lt;=25`</span></span>
<span><span class="co">#&gt; [1] "Cryobacterium roopkundense"  "Hugenholtzia roseola"       </span></span>
<span><span class="co">#&gt; [3] "Halopseudomonas formosensis" "Occallatibacter riparius"   </span></span>
<span><span class="co">#&gt; [5] "Occallatibacter savannae"    "Pectinatus sottacetonis"</span></span></code></pre></div>
<ul>
<li>Create signatures for the animal pathogen attribute (boolean):</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ap_sigs_mix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSignatures.html">makeSignatures</a></span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="va">bp</span><span class="op">[[</span><span class="st">"animal pathogen"</span><span class="op">]</span><span class="op">]</span>, tax_id_type <span class="op">=</span> <span class="st">"NCBI_ID"</span>,</span>
<span>  tax_level <span class="op">=</span> <span class="st">"mixed"</span>, evidence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exp"</span>, <span class="st">"igc"</span>, <span class="st">"nas"</span>, <span class="st">"tas"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">ap_sigs_mix</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`bugphyzz:animal pathogen|FALSE`</span></span>
<span><span class="co">#&gt; [1]  100225 1003110    1006    1008 1008460  101192</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:animal pathogen|TRUE`</span></span>
<span><span class="co">#&gt; [1] 100053 100901   1015   1017   1018   1019</span></span></code></pre></div>
<ul>
<li>Create signatures for all of the bugphyzz data frames:</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">bp</span>, <span class="va">makeSignatures</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/list_flatten.html" class="external-link">list_flatten</a></span><span class="op">(</span>name_spec <span class="op">=</span> <span class="st">"{inner}"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 124</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">sigs</span>, <span class="va">head</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|aerobic`</span></span>
<span><span class="co">#&gt; [1] 1042312  117747    1236 1497346 1562566 1643678</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|anaerobic`</span></span>
<span><span class="co">#&gt; [1]  117743  118001 1313211 1737404  183924  183925</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:aerophilicity|facultatively anaerobic`</span></span>
<span><span class="co">#&gt; [1]    1016 1036132 1069494 1078829  111142  111500</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:antimicrobial resistance|sensitive to amikacin`</span></span>
<span><span class="co">#&gt; [1]    1733    1773     573 1334054 1334056 1334058</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:antimicrobial resistance|sensitive to capreomycin`</span></span>
<span><span class="co">#&gt; [1]    1733    1773 1334054 1334056 1334058 1334062</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`bugphyzz:antimicrobial resistance|sensitive to ethambutol`</span></span>
<span><span class="co">#&gt; [1]    1733    1773 1334054 1334056 1334058 1334062</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-a-bug-set-enrichment-analysis">Run a bug set enrichment analysis<a class="anchor" aria-label="anchor" href="#run-a-bug-set-enrichment-analysis"></a>
</h2>
<p>Bugphyzz signatures are suitable for conducting bug set enrichment
analysis using existing tools available in R. In this example, we will
perform a set enrichment analysis using a dataset with a known
biological ground truth.</p>
<p>The dataset originates from the Human Microbiome Project (2012) and
compares subgingival and supragingival plaque. This data will be
imported using the <a href="https://bioconductor.org/packages/release/data/experiment/html/MicrobiomeBenchmarkData.html" class="external-link">MicrobiomeBenchmarkData
package</a>. For the implementation of the enrichment analysis, we will
utilize the Gene Set Enrichment Analysis (GSEA) method available in the
<a href="https://bioconductor.org/packages/release/bioc/html/EnrichmentBrowser.html" class="external-link">EnrichmentBrowser
package</a>. The expected outcome is an enrichment of aerobic taxa in
the supragingival plaque (positive enrichment score) and anaerobic taxa
in the subgingival plaque (negative enrichment score).</p>
<p>Load necessary packages:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/MicrobiomeBenchmarkData" class="external-link">MicrobiomeBenchmarkData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span></code></pre></div>
<p>Import benchmark data:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_name</span> <span class="op">&lt;-</span> <span class="st">'HMP_2012_16S_gingival_V35'</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">MicrobiomeBenchmarkData</span><span class="fu">::</span><span class="fu"><a href="http://waldronlab.io/MicrobiomeBenchmarkData/reference/getBenchmarkData.html" class="external-link">getBenchmarkData</a></span><span class="op">(</span><span class="va">dat_name</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; adding rname 'https://zenodo.org/record/6911027/files/HMP_2012_16S_gingival_V35_count_matrix.tsv'</span></span>
<span><span class="co">#&gt; adding rname 'https://zenodo.org/record/6911027/files/HMP_2012_16S_gingival_V35_taxonomy_table.tsv'</span></span>
<span><span class="co">#&gt; adding rname 'https://zenodo.org/record/6911027/files/HMP_2012_16S_gingival_V35_taxonomy_tree.newick'</span></span>
<span><span class="co">#&gt; Finished HMP_2012_16S_gingival_V35.</span></span>
<span><span class="va">tse_genus</span> <span class="op">&lt;-</span> <span class="fu">mia</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mia/man/splitByRanks.html" class="external-link">splitByRanks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">$</span><span class="va">genus</span></span>
<span><span class="va">min_n_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">tse_genus</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="va">tse_subset</span> <span class="op">&lt;-</span> <span class="va">tse_genus</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">tse_genus</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_n_samples</span>,<span class="op">]</span></span>
<span><span class="va">tse_subset</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 37 311 </span></span>
<span><span class="co">#&gt; metadata(1): agglomerated_by_rank</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(37): Streptococcus Neisseria ... Mogibacterium Peptococcus</span></span>
<span><span class="co">#&gt; rowData names(7): superkingdom phylum ... genus taxon_annotation</span></span>
<span><span class="co">#&gt; colnames(311): 700103497 700106940 ... 700111586 700109119</span></span>
<span><span class="co">#&gt; colData names(15): dataset subject_id ... sequencing_method</span></span>
<span><span class="co">#&gt;   variable_region_16s</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: a LinkDataFrame (37 rows)</span></span>
<span><span class="co">#&gt; rowTree: 1 phylo tree(s) (45364 leaves)</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p>Perform differential abundance (DA) analysis to get sets of
microbes:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse_subset</span><span class="op">$</span><span class="va">GROUP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span></span>
<span>  <span class="va">tse_subset</span><span class="op">$</span><span class="va">body_subsite</span> <span class="op">==</span> <span class="st">'subgingival_plaque'</span>, <span class="fl">0</span>, <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/deAna.html" class="external-link">deAna</a></span><span class="op">(</span></span>
<span>    expr <span class="op">=</span> <span class="va">tse_subset</span>, de.method <span class="op">=</span> <span class="st">'limma'</span>, padj.method <span class="op">=</span> <span class="st">'fdr'</span>, </span>
<span>    filter.by.expr <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">GROUP</span>, data <span class="op">=</span> <span class="va">dat</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>, design <span class="op">=</span> <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span><span class="op">$</span><span class="va">E</span></span></code></pre></div>
<p>Perform GSEA and display the results:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsea</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/sbea.html" class="external-link">sbea</a></span><span class="op">(</span></span>
<span>  method <span class="op">=</span> <span class="st">'gsea'</span>, se <span class="op">=</span> <span class="va">se</span>, gs <span class="op">=</span> <span class="va">aer_sigs_g</span>, perm <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.1</span> </span>
<span><span class="op">)</span></span>
<span><span class="va">gsea_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gsea</span><span class="op">$</span><span class="va">res.tbl</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    GENE.SET <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">PVAL</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">GENE.SET</span>, <span class="st">' *'</span><span class="op">)</span>, <span class="va">GENE.SET</span><span class="op">)</span>,</span>
<span>    PVAL <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">PVAL</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>BUG.SET <span class="op">=</span> <span class="va">GENE.SET</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">gsea_tbl</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="72%">
<col width="9%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">BUG.SET</th>
<th align="right">ES</th>
<th align="right">NES</th>
<th align="right">PVAL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">bugphyzz:aerophilicity|anaerobic *</td>
<td align="right">-0.877</td>
<td align="right">-1.700</td>
<td align="right">0.004</td>
</tr>
<tr class="even">
<td align="left">bugphyzz:aerophilicity|aerobic *</td>
<td align="right">0.833</td>
<td align="right">1.840</td>
<td align="right">0.004</td>
</tr>
<tr class="odd">
<td align="left">bugphyzz:aerophilicity|facultatively anaerobic</td>
<td align="right">0.268</td>
<td align="right">0.623</td>
<td align="right">0.909</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="get-signatures-associated-with-a-specific-microbe">Get signatures associated with a specific microbe<a class="anchor" aria-label="anchor" href="#get-signatures-associated-with-a-specific-microbe"></a>
</h2>
<p>To retrieve all signature names associated with a specific taxon,
users can utilize the <code>getTaxonSignatures</code> function.</p>
<p>Let’s see an example using <em>Escherichia coli</em> (taxid:
562).</p>
<p>Get all signature names associated to <em>E. coli</em>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getTaxonSignatures.html">getTaxonSignatures</a></span><span class="op">(</span>tax <span class="op">=</span> <span class="st">"Escherichia coli"</span>, bp <span class="op">=</span> <span class="va">bp</span><span class="op">)</span></span>
<span><span class="co">#&gt; character(0)</span></span></code></pre></div>
<p>Get all signature names associated to the <em>E. coli</em> taxid:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getTaxonSignatures.html">getTaxonSignatures</a></span><span class="op">(</span>tax <span class="op">=</span> <span class="st">"562"</span>, bp <span class="op">=</span> <span class="va">bp</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "bugphyzz:aerophilicity|aerobic"                                       </span></span>
<span><span class="co">#&gt;  [2] "bugphyzz:aerophilicity|facultatively anaerobic"                       </span></span>
<span><span class="co">#&gt;  [3] "bugphyzz:arrangement|paired cells"                                    </span></span>
<span><span class="co">#&gt;  [4] "bugphyzz:biosafety level|biosafety level 1"                           </span></span>
<span><span class="co">#&gt;  [5] "bugphyzz:cogem pathogenicity rating|cogem pathogenicity rating 2"     </span></span>
<span><span class="co">#&gt;  [6] "bugphyzz:gram stain|gram stain negative"                              </span></span>
<span><span class="co">#&gt;  [7] "bugphyzz:habitat|digestive system"                                    </span></span>
<span><span class="co">#&gt;  [8] "bugphyzz:habitat|feces"                                               </span></span>
<span><span class="co">#&gt;  [9] "bugphyzz:habitat|human microbiome"                                    </span></span>
<span><span class="co">#&gt; [10] "bugphyzz:shape|bacillus"                                              </span></span>
<span><span class="co">#&gt; [11] "bugphyzz:animal pathogen|FALSE"                                       </span></span>
<span><span class="co">#&gt; [12] "bugphyzz:extreme environment|TRUE"                                    </span></span>
<span><span class="co">#&gt; [13] "bugphyzz:health associated|FALSE"                                     </span></span>
<span><span class="co">#&gt; [14] "bugphyzz:host-associated|TRUE"                                        </span></span>
<span><span class="co">#&gt; [15] "bugphyzz:motility|FALSE"                                              </span></span>
<span><span class="co">#&gt; [16] "bugphyzz:plant pathogenicity|FALSE"                                   </span></span>
<span><span class="co">#&gt; [17] "bugphyzz:spore formation|FALSE"                                       </span></span>
<span><span class="co">#&gt; [18] "bugphyzz:growth temperature|mesophile| &gt; 25 &amp; &lt;= 45"                  </span></span>
<span><span class="co">#&gt; [19] "bugphyzz:mutation rate per site per generation|slow| &gt; 0.78 &amp; &lt;= 2.92"</span></span>
<span><span class="co">#&gt; [20] "bugphyzz:mutation rate per site per year|slow| &gt; 0.08 &amp; &lt;= 7.5"       </span></span>
<span><span class="co">#&gt; [21] "bugphyzz:optimal ph|neutral| &gt; 6 &amp; &lt;= 8"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       Etc/UTC</span></span>
<span><span class="co">#&gt;  date     2024-03-22</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package                 </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version    </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                      1.4-5       <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  annotate                   1.80.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi              1.64.1      <span style="color: #555555;">2023-11-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ape                        5.7-1       <span style="color: #555555;">2023-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  beachmat                   2.18.1      <span style="color: #555555;">2024-02-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  beeswarm                   0.4.0       <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                  * 2.62.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache              2.10.1      <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics             * 0.48.1      <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager                1.30.22     <span style="color: #555555;">2023-08-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors              1.20.2      <span style="color: #555555;">2024-01-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  BiocParallel               1.36.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocSingular               1.18.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle                * 2.30.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               * 2.70.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  bit                        4.0.5       <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bit64                      4.0.5       <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bitops                     1.0-7       <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  blob                       1.2.4       <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bluster                    1.12.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bookdown                   0.38        <span style="color: #555555;">2024-03-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bslib                      0.6.1       <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  bugphyzz                 * 0.0.1.12    <span style="color: #555555;">2024-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  cachem                     1.0.8       <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cli                        3.6.2       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  cluster                    2.1.4       <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  codetools                  0.2-19      <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  colorspace                 2.1-0       <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  crayon                     1.5.2       <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  curl                       5.2.1       <span style="color: #555555;">2024-03-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  DBI                        1.2.2       <span style="color: #555555;">2024-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr                     2.5.0       <span style="color: #555555;">2024-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  DECIPHER                   2.30.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  decontam                   1.22.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedArray               0.28.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedMatrixStats         1.24.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                       1.4.3       <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  digest                     0.6.35      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  DirichletMultinomial       1.44.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  dplyr                    * 1.1.4       <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                      4.0.16      <span style="color: #555555;">2024-02-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  EnrichmentBrowser        * 2.32.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  evaluate                   0.23        <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fansi                      1.0.6       <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                    1.1.1       <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  filelock                   1.0.3       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  fs                         1.6.3       <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  generics                   0.1.3       <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb             * 1.38.8      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData           1.2.11      <span style="color: #555555;">2024-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges            * 1.54.1      <span style="color: #555555;">2023-10-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm                 0.7.2       <span style="color: #555555;">2023-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2                    3.5.0       <span style="color: #555555;">2024-02-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                    0.9.5       <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  glue                       1.7.0       <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  graph                    * 1.80.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  gridExtra                  2.3         <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  GSEABase                   1.64.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  gtable                     0.3.4       <span style="color: #555555;">2023-08-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  htmltools                  0.5.7       <span style="color: #555555;">2023-11-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  httr                       1.4.7       <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  igraph                     2.0.3       <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  IRanges                  * 2.36.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  irlba                      2.3.5.1     <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib                  0.1.4       <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite                   1.8.8       <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  KEGGgraph                  1.62.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  KEGGREST                   1.42.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                      1.45        <span style="color: #555555;">2023-10-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lattice                    0.22-5      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lazyeval                   0.2.2       <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle                  1.0.4       <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  limma                      3.58.1      <span style="color: #555555;">2023-10-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  locfit                     1.5-9.9     <span style="color: #555555;">2024-03-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  magrittr                   2.0.3       <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  MASS                       7.3-60      <span style="color: #555555;">2023-05-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  Matrix                     1.6-1.1     <span style="color: #555555;">2023-09-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics           * 1.14.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats              * 1.2.0       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  memoise                    2.0.1       <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  mgcv                       1.9-0       <span style="color: #555555;">2023-07-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  mia                      * 1.10.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  MicrobiomeBenchmarkData  * 1.4.0       <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  MultiAssayExperiment     * 1.28.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  munsell                    0.5.0       <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  nlme                       3.1-163     <span style="color: #555555;">2023-08-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  permute                    0.9-7       <span style="color: #555555;">2022-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pillar                     1.9.0       <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                  2.0.3       <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                    2.0.7       <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  plyr                       1.8.9       <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  png                        0.1-8       <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  purrr                    * 1.0.2       <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  R6                         2.5.1       <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ragg                       1.3.0       <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                       1.0.12      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                      1.98-1.14   <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  reshape2                   1.4.4       <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  Rgraphviz                  2.46.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  rlang                      1.1.3       <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                  2.26        <span style="color: #555555;">2024-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  RSQLite                    2.3.5       <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                       1.0.5       <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays                   1.2.1       <span style="color: #555555;">2024-03-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  S4Vectors                * 0.40.2      <span style="color: #555555;">2023-11-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  sass                       0.4.9       <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix               1.10.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scales                     1.3.0       <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  scater                     1.30.1      <span style="color: #555555;">2023-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  scuttle                    1.12.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sessioninfo                1.2.2       <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment     * 1.24.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  SparseArray                1.2.4       <span style="color: #555555;">2024-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt;  sparseMatrixStats          1.14.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  statmod                    1.5.0       <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  stringi                    1.8.3       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  stringr                    1.5.1       <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment     * 1.32.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts                1.0.6       <span style="color: #555555;">2024-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  textshaping                0.3.7       <span style="color: #555555;">2023-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tibble                     3.2.1       <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                      1.3.1       <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect                 1.2.1       <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  tidytree                   0.4.6       <span style="color: #555555;">2023-12-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  treeio                     1.26.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  TreeSummarizedExperiment * 2.10.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  utf8                       1.2.4       <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                      0.6.5       <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  vegan                      2.6-4       <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  vipor                      0.4.7       <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  viridis                    0.6.5       <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite                0.4.2       <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  withr                      3.0.0       <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  xfun                       0.42        <span style="color: #555555;">2024-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  XML                        3.99-0.16.1 <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  xtable                     1.8-4       <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  XVector                  * 0.42.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                       2.3.8       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  yulab.utils                0.1.4       <span style="color: #555555;">2024-01-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                   1.48.2      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Samuel Gamboa, Levi Waldron, Kelly Eckenrode, Jonathan Ye, Jennifer Wokaty.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
